vsintegration/src/vs/FsPkgs/FSharp.Project/Common.Source.CSharp/Project/HierarchyNode.cs

```diff
@@ -3301,6 +3300,10 @@ string newTargetFrameworkMoniker
             projectMgr.BuildProject.SetProperty(ProjectFileConstants.TargetFrameworkVersion, HierarchyNode.GetFrameworkVersionString(frameworkName));
             projectMgr.BuildProject.SetProperty(ProjectFileConstants.TargetFrameworkProfile, frameworkName.Profile);
 
+            string targetFSharpCoreVersion = projectMgr.BuildProject.GetPropertyValue(ProjectFileConstants.TargetFSharpCoreVersion);
```

Comment:
Yes it works fine.  For a specific definition of fine.  If the TargetFSharpCore property is missing then the value is "" and so no bindingredirects will be set for the fsharp.core assembly